LFS-1239: start_cards.sh should detect failure to start the java process and terminate

IntegralProgrammer opened the pull request:
This PR causes `start_cards.sh` to exit if the CARDS `java` process terminates. This includes the possibility of the `java` process terminating because the CARDS jar file is not present.

To test:

- Complete these tests in all possible testing environments (WSL, Linux with `python3-psutil` installed, Linux without `python3-psutil` installed)
- After running `mvn clean install`, the `./start_cards.sh` script should work as expected
  - CTRL+C should kill the running CARDS instance
- After running `mvn clean`, running the `./start_cards.sh` script should result in it terminating after a brief period of time as the CARDS jar file is not available
- Rebuild with `mvn clean install`, start CARDS using the `./start_cards.sh` script and kill the CARDS java process using the `kill` command. The `start_cards.sh` script should exit with a _CARDS Java process failure_ message

IntegralProgrammer committed:
LFS-1239: start_cards.sh should detect failure to start the java process and terminate

sdumitriu reviewed (commented):
I don't like relying on the PID, since it is possible for the java process to end and another process to start with the same PID. I also don't like the kill. I think there is another way to check the status of a background process, I'll look it up.

sdumitriu commented on start_cards.sh:
```diff
@@ -132,6 +141,8 @@ then
     #Check if CARDS was able to bind
     python3 Utilities/HostConfig/check_tcp_listen.py --tcp_port $BIND_PORT --pid $CARDS_PID && break
+    #If the CARDS Java process has terminated, stop this script altogether
+    kill -0 $CARDS_PID > /dev/null 2> /dev/null || handle_cards_java_fail
```
This should be in a function since it is used more than once.

A better alternative is to use `jobs` to check if a background job is still active. I think the safest way, is something like this:

```
if [[ -z `jobs -p | grep $CARDS_PID` ]]
then
  echo inactive
else
  echo active
fi
```

This lists the background jobs, and checks if the CARDS_PID is among them. Of course, replace the echos with something more useful.

IntegralProgrammer replied:
Good idea - I have implemented this check using the `jobs` command in commit `adbfed5`

IntegralProgrammer committed:
LFS-1239: start_cards.sh should detect failure to start the java process and terminate

Create a bash function to use the jobs command to check if the CARDS java process is running as opposed to performing this check with kill -0

sashaandjic reviewed (approved):
tested

sashaandjic commented:
mvn clean
![image](https://user-images.githubusercontent.com/23243244/125541509-3832985f-76d0-4276-a285-2308f01762f6.png)

mvn clean install + kill 
![image](https://user-images.githubusercontent.com/23243244/125541533-faa13f13-24e4-427e-8e4e-c99eebe3f3cb.png)
